### PR TITLE
(maint) Source FindRuby

### PR DESCRIPTION
**Pull request by MikaelSmith**
CMake's FindRuby module was missing support for 64-bit Ruby on Windows.
A patch has been submitted; until we move to the patched version, bring
in the patched version.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
Windows CI has been skipping Ruby-related unit tests since I changed how we were using FindRuby.

File: cmake/FindRuby.cmake [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+     set( _RUBY_MSVC_RUNTIME "90" )
+   endif()
+
+   set(_RUBY_ARCH_PREFIX "")
+   if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+     set(_RUBY_ARCH_PREFIX "x64-")
+   endif()
+
+   list(APPEND _RUBY_POSSIBLE_LIB_NAMES
+               "${_RUBY_ARCH_PREFIX}msvcr${_RUBY_MSVC_RUNTIME}-ruby${_RUBY_NODOT_VERSION}"
```
Comment (MikaelSmith) [inline cmake/FindRuby.cmake]: Fix is here, selecting which architecture version to search for. CMAKE_SIZEOF_VOID_P is determined by running the compiler, so it should select appropriately based on the MinGW version found.

File: cmake/FindRuby.cmake [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+               "${_RUBY_ARCH_PREFIX}msvcr${_RUBY_MSVC_RUNTIME}-ruby${_RUBY_NODOT_VERSION}-static"
+               "${_RUBY_ARCH_PREFIX}msvcrt-ruby${_RUBY_NODOT_VERSION}"
+               "${_RUBY_ARCH_PREFIX}msvcrt-ruby${_RUBY_NODOT_VERSION}-static" )
+endif()
+
+find_library(RUBY_LIBRARY NAMES ${_RUBY_POSSIBLE_LIB_NAMES} HINTS ${RUBY_POSSIBLE_LIB_DIR} )
+
+include(FindPackageHandleStandardArgs)
```
Comment (MikaelSmith) [inline cmake/FindRuby.cmake]: Also had to modify this to do lookup correctly.

**Comment by melissa** [comment]
this works for windows 
```
-- The following OPTIONAL packages have been found:

 * Ruby (required version >= 1.9)
```

**Comment by MikaelSmith** [comment]
\o/

**Comment by kylog** [comment]
Woohooooo!

**Pull request merged**